File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):

    def __init__(self):
        super().__init__("py_test")
        self._counter = 0
        self.get_logger().info("hello world from py node!")
        self.create_timer(1.0, self.timer_callback)
    
    def timer_callback(self):
        self.get_logger().info("hello timer aaaa" + str(self._counter))
        self._counter += 1

def main(args=None):
    rclpy.init(args=args) 
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node{
public:
    MyNode() : Node("cpp_test"), _counter(0){
        RCLCPP_INFO(this->get_logger(), "hello world from cpp");
        _timer = this->create_wall_timer(std::chrono::seconds(1), std::bind(&MyNode::timerCallBack, this));
    }
private:
    void timerCallBack(){
        RCLCPP_INFO(this->get_logger(), "Hello timer %d", _counter);
        _counter++;
    }
    rclcpp::TimerBase::SharedPtr _timer;
    int _counter;
};


int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}